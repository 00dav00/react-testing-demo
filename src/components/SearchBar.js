import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

class SearchBar extends Component {
  static propTypes = {
    onSearch: PropTypes.func.isRequired,
  }

  state = { searchText: '' }

  render() {
    return (
      <Bar>
        <SearchInput 
          id="search-input"
          onChange={e => this.setState({ searchText: e.target.value })}
        />
        <SearchButton
          id="search-button"
          onClick={() => this.props.onSearch(this.state.searchText)}
        >
          Search
        </SearchButton>
      </Bar>
    );
  }
}

const Bar = styled.div`
  height: 40px;
  width: 300px;
  display: flex;
  margin-top: 1em;
`

const SearchInput = styled.input`
  width: 200px;
  height: 100%;
  padding: 0;
  margin: 0;
  margin-right: 1em;
  background-color: transparent;
  border: 2px solid #3399FF;
  border-radius: 5px;
  box-sizing: border-box;
  text-align: center;
  outline: none;
`;

const SearchButton = styled.button`
  width: 100px;
  height: 100%;
  background-color: #3399FF;
  border: 2px solid #3399FF;
  border-radius: 5px;
  color: white;
  cursor: pointer;
`;

export default SearchBar;