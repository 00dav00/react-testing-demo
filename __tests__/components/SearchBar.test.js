import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '../../src/components/SearchBar';

describe('Test SearchBar component', () => {
  it('Should render', () => {
    const component = renderer.create(
      <SearchBar />
    );
  
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
});