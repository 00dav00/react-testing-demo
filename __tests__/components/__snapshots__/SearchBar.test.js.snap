// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchBar component Should render 1`] = `
<div
  className="sc-bdVaJa isQJlR"
>
  <input
    className="sc-bwzfXH izmQdJ"
  />
</div>
`;
